include/deal.II/grid/manifold_lib.h

```diff
@@ -383,11 +383,26 @@ class CylindricalManifold : public ChartManifold<dim,spacedim,3>
    * through the given #point_on_axis. The direction may be arbitrarily
    * scaled, and the given point may be any point on the axis. The tolerance
    * value is used to determine if a point is on the axis.
+   *
+   * @deprecated Use the constructor that takes a Tensor as the first argument
+   * instead.
    */
+  DEAL_II_DEPRECATED
   CylindricalManifold (const Point<spacedim> &direction,
```

Comment:
Just remove this constructor outright -- there is no compatibility issue for constructors because a `Point` automatically converts to `Tensor` if you chose to pass in a `Point`. For regular member functions, there is a potential incompatibility if you tried to take the address of a member function and assign it to a pointer-to-member that now would have to have a different type. But you can't take the address of a constructor, so this is not an issue.

Reply:
This is much nicer than using `static_cast`. I'll update the PR.